echopype/utils/coding.py

```diff
@@ -142,16 +144,19 @@ def set_zarr_encodings(ds: xr.Dataset, compression_settings: dict) -> dict:
     # create zarr specific encoding
     encoding = dict()
     for name, val in ds.variables.items():
-        val_encoding = val.encoding
-        val_encoding.update(get_zarr_compression(val, compression_settings))
+        encoding[name] = {**val.encoding}
+        encoding[name].update(get_zarr_compression(val, compression_settings))
 
-        # If data array is not a dask array yet,
-        # create a custom chunking encoding
-        # currently defaults to 100MB
-        if not ds.chunks and len(val.shape) > 0:
+        # Always optimize chunk if not specified already
+        # user can specify desired chunk in encoding
+        existing_chunks = encoding[name].get("chunks", None)
+        if len(val.shape) > 0 and existing_chunks is not None:
             chunks = _get_auto_chunk(val)
-            val_encoding.update({"chunks": chunks})
-        encoding[name] = val_encoding
+            encoding[name]["chunks"] = chunks
```

Comment:
Should the second of the if statement be `existing_chunks is None` ?
The way it is if `existing_chunks` is given, chunks from `_get_auto_chunk` will be used, which seems the opposite of what should be happening?

Should we have a simple test for this function?

Reply:
Just updated this to be a little smarter, where it checks the existing chunk as to whether it's optimal or not! If not optimal then run the auto chunk. I wanna make sure it's always optimal especially as data get's larger. At some point we'll think about letting user choose the chunk size, but right now, we're "optimizing" the chunk size for simplicity sake.